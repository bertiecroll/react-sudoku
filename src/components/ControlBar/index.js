export { ControlBar } from './ControlBar'
export { default } from './ControlBar'
