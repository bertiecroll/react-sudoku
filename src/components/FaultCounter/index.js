export { FaultCounter } from './FaultCounter'
export { default } from './FaultCounter'
