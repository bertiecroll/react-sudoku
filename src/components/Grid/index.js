export { Grid } from './Grid'
export { GridRow } from './GridRow'
