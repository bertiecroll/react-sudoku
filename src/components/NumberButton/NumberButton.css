.NumberButton {
  width: 2em;
  height: 2em;
  margin: .15em;
  line-height: 2em;
  font-size: 1.5em;
  text-align: center;
  border-radius: 5px;
  box-shadow: 1px 1px 1px 0px #000;
}

.NumberButton:nth-child(odd) {
  transform: rotate(5deg);
}

.NumberButton:nth-child(even) {
  transform: rotate(-5deg);
}

.NumberButton:hover {
  transform: scale(1.1);
  transition: transform .7s;
  cursor: pointer;
}

.not-available {
  background-color: grey;
  opacity: 0.2;
}

.not-available:hover {
  transform: none;
  cursor: not-allowed;
}
