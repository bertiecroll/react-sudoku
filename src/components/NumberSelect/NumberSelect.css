.NumberSelect {
  margin: 1em;
  padding: .25em;
  display: flex;
  flex-direction: row;
}
