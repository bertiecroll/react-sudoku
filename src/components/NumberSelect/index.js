export { NumberSelect } from './NumberSelect'
export { default } from './NumberSelect'
