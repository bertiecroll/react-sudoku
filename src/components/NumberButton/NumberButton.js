import React from 'react'
import PropTypes from 'prop-types'

import './NumberButton.css'

export default function NumberButton ({ number, isAvailable, onClickHandler }) {
  const classNames = `NumberButton ${isAvailable ? '' : 'not-available'}`

  return (
    <div
      className={classNames}
      onClick={onClickHandler}>
      {number}
    </div>
  )
}

NumberButton.defaultProps = {
  isAvailable: false,
  onClickHandler: () => {},
}

NumberButton.propTypes = {
  number: PropTypes.number.isRequired,
  isAvailable: PropTypes.bool.isRequired,
  onClickHandler: PropTypes.func.isRequired,
}
