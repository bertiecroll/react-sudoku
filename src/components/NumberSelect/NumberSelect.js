import React from 'react'
import PropTypes from 'prop-types'
import './NumberSelect.css'

export const NumberSelect = (props) => {
  return (
    <div className="NumberSelect">
      {props.availableNumbers.map(number => <div key={number} onClick={() => props.handleClick(number)}>{number}</div>)}
    </div>
  )
}

NumberSelect.propTypes = {
  availableNumbers: PropTypes.array.isRequired
}
