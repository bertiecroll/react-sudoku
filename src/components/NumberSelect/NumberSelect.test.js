import React from 'react'
import { shallow } from 'enzyme'
import { NumberSelect } from 'components/NumberSelect'
import NumberButton from 'components/NumberButton'

describe('PencilMarker', () => {
  const mockOnClickAction = jest.fn()

  afterEach(() => {
    mockOnClickAction.mockClear()
  })

  it('renders NumberButton component for each number in allNumbers array', () => {
    const wrapper = shallow(
      <NumberSelect
        allNumbers={[1, 2, 3, 4, 5, 6, 7, 8, 9]}
        availableNumbers={[1, 3, 5, 7, 9]}
        onClickAction={mockOnClickAction}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  describe('#onClickHandler', () => {
    it('calls #onClickAction when selectedCell given and completed is false', () => {
      const selectedCell = { id: '0,0', value: 1 }
      const selectedNumber = 1
      const wrapper = shallow(
        <NumberSelect
          selectedCell={selectedCell}
          allNumbers={[1, 2, 3, 4, 5, 6, 7, 8, 9]}
          availableNumbers={[1, 3, 5, 7, 9]}
          onClickAction={mockOnClickAction}
        />
      )

      wrapper.find(NumberButton).first().props().onClickHandler(selectedNumber)

      expect(mockOnClickAction).toHaveBeenCalledWith(selectedCell, selectedNumber)
    })

    it('does not call #onClickAction when selectedCell is undefined', () => {
      const selectedNumber = 1
      const wrapper = shallow(
        <NumberSelect
          allNumbers={[1, 2, 3, 4, 5, 6, 7, 8, 9]}
          availableNumbers={[1, 3, 5, 7, 9]}
          onClickAction={mockOnClickAction}
        />
      )

      wrapper.find(NumberButton).first().props().onClickHandler(selectedNumber)

      expect(mockOnClickAction).not.toHaveBeenCalled()
    })

    it('does not call #onClickAction when selectedCell is completed', () => {
      const selectedCell = { id: '0,0', value: 1, completed: true }
      const selectedNumber = 1
      const wrapper = shallow(
        <NumberSelect
          selectedCell={selectedCell}
          allNumbers={[1, 2, 3, 4, 5, 6, 7, 8, 9]}
          availableNumbers={[1, 3, 5, 7, 9]}
          onClickAction={mockOnClickAction}
        />
      )

      wrapper.find(NumberButton).first().props().onClickHandler(selectedNumber)

      expect(mockOnClickAction).not.toHaveBeenCalled()
    })
  })
})
