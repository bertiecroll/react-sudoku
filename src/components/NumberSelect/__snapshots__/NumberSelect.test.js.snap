// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PencilMarker renders NumberButton component for each number in allNumbers array 1`] = `
<div
  className="NumberSelect"
>
  <NumberButton
    isAvailable={true}
    key="1"
    number={1}
    onClickHandler={[Function]}
  />
  <NumberButton
    isAvailable={false}
    key="2"
    number={2}
    onClickHandler={null}
  />
  <NumberButton
    isAvailable={true}
    key="3"
    number={3}
    onClickHandler={[Function]}
  />
  <NumberButton
    isAvailable={false}
    key="4"
    number={4}
    onClickHandler={null}
  />
  <NumberButton
    isAvailable={true}
    key="5"
    number={5}
    onClickHandler={[Function]}
  />
  <NumberButton
    isAvailable={false}
    key="6"
    number={6}
    onClickHandler={null}
  />
  <NumberButton
    isAvailable={true}
    key="7"
    number={7}
    onClickHandler={[Function]}
  />
  <NumberButton
    isAvailable={false}
    key="8"
    number={8}
    onClickHandler={null}
  />
  <NumberButton
    isAvailable={true}
    key="9"
    number={9}
    onClickHandler={[Function]}
  />
</div>
`;
